std/digest/murmurhash.d

```diff
@@ -0,0 +1,753 @@
+            h2 = update(h2, block[1], h1, c2, c1, 33, 31, 0x38495ab5U);
+        }
+
+        /// Put remainder bytes. This must be called only once after `putElement` and before `finalize`.
+        void putRemainder(scope const(ubyte[]) data...) pure nothrow @nogc
+        {
+            assert(data.length < Element.sizeof);
+            assert(data.length >= 0);
+            element_count += data.length;
+            ulong k1 = 0;
+            ulong k2 = 0;
+            final switch (data.length & 15)
+            {
+            case 15:
+                k2 ^= ulong(data[14]) << 48;
+                goto case;
+            case 14:
+                k2 ^= ulong(data[13]) << 40;
+                goto case;
+            case 13:
+                k2 ^= ulong(data[12]) << 32;
+                goto case;
+            case 12:
+                k2 ^= ulong(data[11]) << 24;
+                goto case;
+            case 11:
+                k2 ^= ulong(data[10]) << 16;
+                goto case;
+            case 10:
+                k2 ^= ulong(data[9]) << 8;
+                goto case;
+            case 9:
+                k2 ^= ulong(data[8]) << 0;
+                h2 ^= shuffle(k2, c2, c1, 33);
+                goto case;
+            case 8:
+                k1 ^= ulong(data[7]) << 56;
+                goto case;
+            case 7:
+                k1 ^= ulong(data[6]) << 48;
+                goto case;
+            case 6:
+                k1 ^= ulong(data[5]) << 40;
+                goto case;
+            case 5:
+                k1 ^= ulong(data[4]) << 32;
+                goto case;
+            case 4:
+                k1 ^= ulong(data[3]) << 24;
+                goto case;
+            case 3:
+                k1 ^= ulong(data[2]) << 16;
+                goto case;
+            case 2:
+                k1 ^= ulong(data[1]) << 8;
+                goto case;
+            case 1:
+                k1 ^= ulong(data[0]) << 0;
+                h1 ^= shuffle(k1, c1, c2, 31);
+                goto case;
+            case 0:
+            }
+        }
+
+        /// Incorporate `element_count` and finalizes the hash.
+        void finalize() pure nothrow @nogc
+        {
+            h1 ^= element_count;
+            h2 ^= element_count;
+
+            h1 += h2;
+            h2 += h1;
+            h1 = fmix(h1);
+            h2 = fmix(h2);
+            h1 += h2;
+            h2 += h1;
+        }
+
+        /// Returns the hash as an ulong[2] value.
+        Element get() pure nothrow @nogc
+        {
+            return [h1, h2];
+        }
+
+        /// Returns the current hashed value as an ubyte array.
+        ubyte[16] getBytes() pure nothrow @nogc
+        {
+            return cast(typeof(return)) get();
+        }
+    }
+    else
+    {
+        import std.conv;
+
+        static assert(false,
+                "MurmurHash3(" ~ size.to!string ~ "," ~ opt.to!string ~ ") is not implemented");
```

Comment:
Do not need to import `std.conv`: `size.stringof` and `opt.stringof` should work